int32 state
-----
bool message